File: trustdevice/src/main/java/cn/tongdun/mobrisk/core/collectors/MemoryInfoCollector.java
package cn.tongdun.mobrisk.core.collectors;

import android.app.ActivityManager;
import android.os.Environment;
import android.os.StatFs;

import java.io.File;

/**
 * @description: Memory Info
 * @author: wuzuchang
 * @date: 2022/12/6
 */
public class MemoryInfoCollector {

    private long mAvailableMemory;
    private long mTotalMemory;
    private long mAvailableStorage;
    private long mTotalStorage;

    public MemoryInfoCollector(ActivityManager activityManager) {
        getMemoryInfo(activityManager);
        getStorageInfo();
    }

    private void getMemoryInfo(ActivityManager activityManager) {
        try {
            ActivityManager.MemoryInfo memoryInfo = new ActivityManager.MemoryInfo();
            activityManager.getMemoryInfo(memoryInfo);
            mAvailableMemory = memoryInfo.availMem;
            mTotalMemory = memoryInfo.totalMem;
        } catch (Exception ignored) {
        }
    }

    public void getStorageInfo() {
        try {
            File path = Environment.getDataDirectory();
            StatFs stat = new StatFs(path.getPath());
            long blockSize = stat.getBlockSizeLong();
            long availableBlocks = stat.getAvailableBlocksLong();
            long totalBlocks = stat.getBlockCountLong();
            mTotalStorage = totalBlocks * blockSize;
            mAvailableStorage = availableBlocks * blockSize;
        } catch (Exception ignored) {
        }
    }

    public long getTotalMemory() {
        return mTotalMemory;
    }

    public long getAvailableMemory() {
        return mAvailableMemory;
    }

    public long getAvailableStorage() {
        return mAvailableStorage;
    }

    public long getTotalStorage() {
        return mTotalStorage;
    }
}

File: trustdevice/src/main/java/cn/tongdun/mobrisk/core/collectors/MemoryInfoCollector.kt
package cn.tongdun.mobrisk.core.collectors

import android.app.ActivityManager
import android.os.Environment
import android.os.StatFs
import cn.tongdun.mobrisk.core.tools.executeSafe

/**
 * @description:MemoryInfo
 * @author: wuzuchang
 * @date: 2023/5/15
 */
interface MemoryInfoInterface {
    fun getTotalMemory(): Long
    fun getAvailableMemory(): Long
    fun getAvailableStorage(): Long
    fun getTotalStorage(): Long
}

class MemoryInfoCollector(private val activityManager: ActivityManager) : MemoryInfoInterface {
    private var mAvailableMemory = 0L
    private var mTotalMemory = 0L
    private var mAvailableStorage = 0L
    private var mTotalStorage = 0L

    init {
        executeSafe {
            val memoryInfo = ActivityManager.MemoryInfo()
            activityManager.getMemoryInfo(memoryInfo)
            mAvailableMemory = memoryInfo.availMem
            mTotalMemory = memoryInfo.totalMem

            val path = Environment.getDataDirectory()
            val stat = StatFs(path.path)
            val blockSize = stat.blockSizeLong
            val availableBlocks = stat.availableBlocksLong
            val totalBlocks = stat.blockCountLong
            mTotalStorage = totalBlocks * blockSize
            mAvailableStorage = availableBlocks * blockSize
        }
    }

    override fun getTotalMemory(): Long {
        return mTotalMemory
    }

    override fun getAvailableMemory(): Long {
        return mAvailableMemory
    }

    override fun getAvailableStorage(): Long {
        return mAvailableStorage
    }

    override fun getTotalStorage(): Long {
        return mTotalStorage
    }

}